### PR TITLE
Prevent infinite loop within Command init()

### PR DESCRIPTION
**Pull request by kevingelion**
While debugging a recent install of 2.beta, @anulman and I came across a strange loop within `-command.js`:

### CODE REVIEW

File: lib/commands/-command.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = Command.extend({
   init() {
     process.env.EMBER_CLI_ELECTRON = true;
 
-    if (this._super && this._super.init) {
+    if (this._super && this._super.init && this._super.init !== this.init) {
```
Comment (bendemboski) [inline lib/commands/-command.js]: Can't this whole block just be `this._super(...arguments)` (or `this._super.apply(this, arguments)` if we care about older Node compatibility)?

Comment (bendemboski) [inline lib/commands/-command.js]: That's what I've been doing [here](https://github.com/felixrieseberg/ember-electron/blob/master/lib/commands/electron-test.js#L12), and it's working fine AFAIKT.

Comment (anulman) [inline lib/commands/-command.js]: That would also be my preference; I vaguely remember trying that at one point but there was a lot of churn so ¯\_(ツ)_/¯

Comment (kevingelion) [inline lib/commands/-command.js]: Ideally, I would also prefer to use your suggestions but before making this PR i tried using ES6 syntax (identical to what you're doing @bendemboski) and was met with:
```Function.prototype.apply was called on #<Class>, which is a object and not a function
TypeError: Function.prototype.apply was called on #<Class>, which is a object and not a function```

**Comment by kevingelion** [comment]
Worthwhile to note that I also could reproduce this on Windows x64.

File: lib/commands/-command.js (hunk 1 above)
Comment (bendemboski) [inline lib/commands/-command.js]: @kevingelion are you sure you were running against the correct ember-cli version? When I change to that syntax, everything works fine for me locally. Do you have a more specific repro?

If this is a problem here, then it's probably a problem in `electron-test.js` also, so we probably want to fix it in both places, but I'd prefer to understand why this pretty fundamental mechanism of ember-cli's `CoreObject` is misbehaving for us...

Comment (anulman) [inline lib/commands/-command.js]: Total conjecture: could my issues have stemmed from using an older e-cli version; and @kevingelion's from keeping the `.apply(...)` syntax?

Comment (bendemboski) [inline lib/commands/-command.js]: Plausible -- I don't remember exactly how, but I'm pretty sure in older version of ember-cli invoking super-class functions was more fiddly.

Comment (kevingelion) [inline lib/commands/-command.js]: Ah, okay. I'm running ember-cli `2.4.3`. My fault for not even considering that...

**Comment by anulman** [comment]
@kevingelion I pushed the change discussed inline & am about to merge. If the behaviour recurs, we know what to do :)

**Pull request merged**